manifests/instance.pp

```diff
@@ -112,6 +118,14 @@
     $instance_configdir = $configdir
   }
 
+  # manage_dir setting
```

Comment:
You can get rid of this logic now with the modification you made for the default value :-)